openpathsampling/experimental/storage/test_monkey_patches.py

```diff
@@ -0,0 +1,43 @@
+import openpathsampling as paths
+import pytest
+
+from openpathsampling.experimental.storage.monkey_patches import *
+from openpathsampling.experimental.storage.collective_variables import \
+    CollectiveVariable
+
+@pytest.fixture
+def patching_tps_network():
+    cv = CollectiveVariable(lambda x: x.xyz[0][0]).named('cv')
+    state_A = paths.CVDefinedVolume(cv, 0, 1)
+    state_B = paths.CVDefinedVolume(cv, 9, 10)
+    tps_network = paths.TPSNetwork(state_A, state_B)
+    yield tps_network
+
+
+def test_double_patch(patching_tps_network):
+    orig_dict = patching_tps_network.to_dict()
+    global paths
+    paths = monkey_patch_all(paths)
+    try:
+        patched_dict = patching_tps_network.to_dict()
+        paths = monkey_patch_all(paths)
+        double_patched_dict = patching_tps_network.to_dict()
+    finally:
+        unpatch(paths)
+
+    assert orig_dict != patched_dict
+    assert patched_dict == double_patched_dict
```

Comment:
I would probably rewrite this so that the whole test is caught in the try (including the asserts) as now 
`double_patched_dict` is not guaranteed to be defined if the `monkey_path_all` fails.

I would probably rewrite like:
```suggestion
    try:
        patched_dict = patching_tps_network.to_dict()
        paths = monkey_patch_all(paths)
        double_patched_dict = patching_tps_network.to_dict()
        assert orig_dict != patched_dict
        assert patched_dict == double_patched_dict
    except:
        unpatch(paths)
        raise
    else:
        unpatch(paths)
```

Reply:
I'm not concerned by this -- the `finally` clause doesn't prevent the error from raising. So if there's an error, the asserts should not be reached.